src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,39 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int ARRAY_MAX_SIZE = 10;
+    private int size = 0;
+
+    private final Object[][] resultArray = new Object[ARRAY_MAX_SIZE][2];
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < size; i++) {
+            if (key == null && resultArray[i][0] == null
+                    || key != null && key.equals(resultArray[i][0])) {
+                resultArray[i][0] = key;
+                resultArray[i][1] = value;
```

Comment:
```suggestion
                resultArray[i][1] = value;
```
ключі однакові їх не потрібно ще раз сетити